### ISSUE TITLE
Option to Flat Rectangle Material Thumbnails

### ISSUE THREAD

**Issue by gvarados1** [Facepunch/sbox-issues#1874]
### What it is?

We can't preview materials properly with a sphere thumbnail. We can't see the fine details for materials like trims, bricks, or anything that isn't noise (grass, bare concrete)

Environment Art in hammer consists of a lot of fine UV adjustment. We need to be able to see the brick patterns, trim arrangements, and the top/bottoms of things like concrete. We're not just placing one material down on a surface and forgetting about.

Sphere thumbnails work in other engines like Unity because they don't have proper level design tools. The most you're doing with a material is applying it to an entire surface or terrain. Any advanced UV work is done in another tool and imported as a model.

**Here are some examples:**
With a flat preview we can clearly see all the trim patterns and how they're tiled
The sphere preview is just mush and we can't see what's going on.
![image](https://user-images.githubusercontent.com/39461072/170843576-f4b0a780-6916-4eeb-8dc0-594475ac34c6.png)

This material was originally intended for a model, however, it has a lot of small unique details which can be used on trims. With a sphere thumbnail, we wouldn't be able to see all the details and we'd never know that we can use it.
![image](https://user-images.githubusercontent.com/39461072/170843461-4b75bd97-e2af-4225-8acb-096a3325e30f.png)

I don't know what this material is. It's a complete mystery especially with the backface in the preview. We'd be able to clearly see it with a flat preview.
![image](https://user-images.githubusercontent.com/39461072/170843619-81bb6d27-7b02-426b-bc47-efef4fd4253e.png)

### What should it be?

This change should either be reverted or there should be an option to use a flat preview.

**Additional context**

We also should have the grid returned to model thumbnails. It's not as important as this but it's thrown me off more than a few times. It's impossible to tell the scale of a model without a reference (ex. grid) #1550

Artists need to be consulted before making changes like this. These are not just cosmetic changes. It's negatively affecting workflow.

**Comment by gvarados1** [comment]
Sphere thumbnails have some use cases. I think they're a lot more pleasant looking in-game. Maybe materials should generate both a flat & sphere thumbnail for different use cases.

**Comment by JammieDodgerr** [comment]
If the sphere was rotated towards the camera (or vice versa) it would help a lot with the readability.

![before](https://user-images.githubusercontent.com/60801006/170844554-669afb9f-817b-45e7-adff-625ff0b1ce6e.png)
![after](https://user-images.githubusercontent.com/60801006/170844555-3190a7b7-3298-4127-b6d8-afad0e9f247a.png)

**Comment by garrynewman** [comment]
We programmatically choose a sphere or plane thumbnail. Right now it uses a sphere if the material has a `DefaultMaterialPreviewToSphere` attribute set to non zero, or it has a `VrForward` pass. I believe that the sphere should be the default choice for 90% of materials, since it shows normal map, specular etc. better.

But the selection could do with more finessing.. I'd rather try to be able to choose automatically rather than making it a global choice.. so maybe we can come up with some rules the thumbnailing system could use, based on the material name and contents?

**Comment by KayaDLX** [comment]
I really think a global option across all s&box tools to toggle between flat and sphere previews would be much better than per material with the thumbnailing system.

As someone that uses Hammer a lot, spheres just make it much harder to tell the details especially when you work with hotspots or squared materials. Zooming out and distorting the materials is a downgrade when choosing a texture to work with.

I'd rather want to pick a material because of how it looks rather than how the normal and specular look. (and if I remember correctly, flat previews already showed specular with an overhead light on previews?)

Spheres are good for showcases when you want to show off specular, but not really useful in practice IMO.

**Comment by bakscratch** [comment]
My main concern is it makes world materials unreadable at a glance, So examples;
![image](https://user-images.githubusercontent.com/8184184/170866790-47379635-f471-4953-9eca-3e2cf9f96eae.png)

![image](https://user-images.githubusercontent.com/8184184/170866833-c6d4022d-8124-4140-8fd9-59fb5084720e.png)

As to what Kaya said, the old way showed all the information pretty clearly (as seen in the image above)

The distortion on the sphere also makes it hard to see trim sheets and hotspots correctly as seen here;
![image](https://user-images.githubusercontent.com/8184184/170866958-220d5588-7b30-4a9b-ba09-ca012bea0b7a.png)

I think its fine in other engines as they mostly rely on using static meshes for level design so having the materials not be as readable is ok, where as S2 Hammer the materials are used a lot more in the world geo and being readable helps mappers a lot more. (This isn't just because Source Engine brain its a differen't workflow to said other engines)

**Comment by wouterpleizier** [comment]
Yeah, the base texure and UV layout are far more valuable to me as a mapper. In the rare cases where things like the roughness or normal mapping would be the deciding factor, I'd be more inclined to quickly apply potential materials to my geometry so I can see how they actually interact with the lighting in my map. In this situation, a spherical preview with some generic lighting scenario does not benefit me any more than the old behaviour already did.

> Artists need to be consulted before making changes like this. These are not just cosmetic changes. It's negatively affecting workflow.

I share your frustration. Changing things so they work the same way as the majority of other tools does not make much sense when the majority of other tools aren't designed around level design and environment art.

**Comment by KayaDLX** [comment]
I could not word it any better.

**Comment by KingOfTheFBI** [comment]
Just make option for square or ball. Like a little check mark
